main.js

```diff
@@ -12,9 +12,46 @@ window.addEventListener('load', trigger.bind(null, 'load'));
 window.addEventListener('load', trigger.bind(null, 'DOMContentLoaded'));
 document.addEventListener('DOMContentLoaded', trigger.bind(null, 'DOMContentLoaded'));
 
+document.onreadystatechange = function () {
+	if (document.readyState === 'complete') {
+		trigger('DOMContentLoaded');
+		trigger('load');
+	} else if (document.readyState === 'interactive' && !document.attachEvent) {
+		trigger('DOMContentLoaded');
+	}
+};
+
 if (document.readyState === 'complete') {
-	trigger('load');
 	trigger('DOMContentLoaded');
-} else if (document.readyState === 'interactive') {
+	trigger('load');
+} else if (document.readyState === 'interactive' && !document.attachEvent) {
 	trigger('DOMContentLoaded');
 }
+
+
+if (document.attachEvent) {
+	// If IE and not a frame
+	// continually check to see if the document is ready
+	var top = false;
+
+	try {
+		top = window.frameElement == null && document.documentElement;
+	} catch(e) {}
+
+	if (top && top.doScroll) {
+		(function doScrollCheck() {
+			if (!('DOMContentLoaded' in hasFired)) {
+
+				try {
+					// Use the trick by Diego Perini
+					// http://javascript.nwbox.com/IEContentLoaded/
+					top.doScroll("left");
+				} catch(e) {
+					return setTimeout( doScrollCheck, 50 );
```

Comment:
Can we maybe have some backoff and limit on this?  Say you define `delay` and initialise it to 50, then:

```
return delay < 5000 ? setTimeout( doScrollCheck, (delay*=1.2)) : undefined;
```

So delay increases by 20% each iteration until it's more than 5s, at which point it gives up.  That would take 34 seconds in 26 iterations, so you could obviously play around with the numbers a bit.  I used this code to work it out:

```
var delay = 50, i=[]; while (delay<5000) i.push(delay*=1.2); console.log(i.length, i.reduce(function(sum, el) { sum += el; return sum; }, 0));
```